types/helpers.d.ts

```diff
@@ -1,32 +1,52 @@
 import Vue = require("vue");
 
 type Dictionary<T> = { [key: string]: T };
+type Computed = () => any;
+type MutationMethod = (...args: any[]) => void;
+type ActionMethod = (...args: any[]) => Promise<any>;
 
-export function mapState (map: string[]): Dictionary<() => any>;
-export function mapState (namespace: string, map: string[]): Dictionary<() => any>;
-export function mapState (map: Dictionary<string>): Dictionary<() => any>;
-export function mapState (namespace: string, map: Dictionary<string>): Dictionary<() => any>;
-export function mapState <S>(
-  map: Dictionary<(this: typeof Vue, state: S, getters: any) => any>
-): Dictionary<() => any>;
-export function mapState <S>(
-  namespace: string,
-  map: Dictionary<(this: typeof Vue, state: S, getters: any) => any>
-): Dictionary<() => any>;
+type Mapper<R> = {
```

Comment:
I would prefer using interface as long as possible.

It enables module augmentations, and also is more performant when compiling.